### DISCUSSION TITLE
VSCode TypeScript IntelliSense slowdown issue

### DISCUSSION THREAD

**Discussion by kadirgun** [icflorescu/mantine-datatable#597] [Q&A]
When using mantine-datatable in my project vscode tsserver is extremely slow. Intellisense suggestions sometimes take 2-3 seconds to load.

https://github.com/icflorescu/mantine-datatable/assets/1864516/81b54e7a-6349-4a89-afdf-39d75ed7b2b7

On any file without a datatable component, tsserver works fast.

https://github.com/icflorescu/mantine-datatable/assets/1864516/fb9871ff-9363-4195-9d3d-282c23c5d3e4

Is anyone else having this problem? How can I solve it?

**Comment by riccardocalligaro** [comment]
Facing the same problem here. It slows down the entire file and it can take even 5/10 seconds for errors or autosuggestions to appear. 
I have an 2020 Macbook Pro with an i9 and 16GB of RAM.

**Comment by wachidmudi** [comment]
I also having the same issue on MBP M1 and 16GB of RAM, i thought it was some of the installed extension that caused it, but yeah the issue occurred after installing `"mantine-datatable": "^7.11.3"` i believe

**Comment by vvel0x** [comment]
I am facing the same issue on my editor. For context, I work on a remote machine through SSH with 32GB of RAM and super fast NVMe SSDs (Kioxia CD6s). Hopefully someone figures out a way to resolve this. Although not a deal breaker, the performance penalty does make working with data tables rather annoying. 

EDIT: It appears someone has come up with a temporary solution to this in issue #651

**Comment by stefan-girlich** [comment]
Similar issue, using MBP M2 Max, 96GB and `"mantine-datatable": "^7.15.1"`.
IntelliSense fails to respond and shows an inline loading spinner:
![Screenshot 2025-04-24 at 15 16 26](https://github.com/user-attachments/assets/ab8d4a5c-1c34-419f-b9b6-dfabda3cc6d8)

I just tried to paste into a line of code as a prop to `<DataTable />`: The inline spinner appeared for 20 seconds, then disappeared without the line being pasted.

When I activate the `vimvscode` extension, the editor freezes indefinitely and does not respond do keyboard input anymore.
